File: react_native_app/src/core/error/failures.ts
// Fallos de dominio. La capa `data` traduce las excepciones técnicas
// (`ServerException`, `NetworkException`) a estos fallos, y la capa
// `presentation` los muestra como mensajes claros al usuario.

export abstract class Failure extends Error {
  constructor(message: string) {
    super(message);
    this.name = this.constructor.name;
  }
}

export class ServerFailure extends Failure {
  constructor(message = 'Ocurrió un error en el servidor.') {
    super(message);
  }
}

export class NetworkFailure extends Failure {
  constructor(message = 'Sin conexión. Revisa tu red e intenta de nuevo.') {
    super(message);
  }
}

File: flutter_app/lib/core/error/failures.dart
/// Fallos de dominio. La capa `data` traduce las excepciones técnicas
/// (`ServerException`, `NetworkException`) a estos fallos, y la capa
/// `presentation` los muestra como mensajes claros al usuario.
sealed class Failure implements Exception {
  final String message;
  const Failure(this.message);

  @override
  String toString() => message;
}

class ServerFailure extends Failure {
  const ServerFailure([super.message = 'Ocurrió un error en el servidor.']);
}

class NetworkFailure extends Failure {
  const NetworkFailure([super.message = 'Sin conexión. Revisa tu red e intenta de nuevo.']);
}
